src/etc/inc/interfaces.inc

```diff
@@ -3039,14 +3039,44 @@ function find_dhcp6c_process($interface) {
 	return intval($pid);
 }
 
-function kill_dhcp6client_process($interface) {
+function kill_dhcp6client_process($interface,$norelease) {
+	global $g;
+
 	if (empty($interface) || !does_interface_exist($interface)) {
 		return;
 	}
 
 	if (($pid = find_dhcp6c_process($interface)) != 0) {
-		mwexec("kill -9 {$pid}");
-		sleep(1);
+		/* Kill -9 caused the pid to get left behind, also if we need a */
+		/* relase sent then it needs to be -15, this then allows dhcp6c */
+		/* to send the release, it will also clean up after itself */
+		// Debug - next line remove if not required
+		mwexec("/usr/bin/logger -t info 'shutting down dhcp6c process'");
+		$killtype = isset($norelease) ? "-9" : "-15";
+		mwexec("kill {$killtype} {$pid}"); 
```

Comment:
It would, but several months ago it was found posix_kill() was not killing dhcp6c as it should, hence it being the way it is and has been for quite a while.